.github/workflows/build.yaml

```diff
@@ -3,14 +3,39 @@ name: Build library
 on: [ push ]
 
 jobs:
+  mock-build:
+    name: Ensure the build works
+    runs-on: ubuntu-latest
+
+    if: github.ref != 'refs/heads/main'  # Only upload if change was to the main branch
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python environment `3.10`
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r "requirements.dev.txt"
+
+      - name: Build
+        run: python -m build
+
   publish-and-publish:
```

Comment:
you could also make this a separate WF and trigger it only on main

Reply:
Ya, I think I like how this logically groups it though, keeps the related things together